src/hotspot/share/opto/c2compiler.cpp

```diff
@@ -216,6 +216,10 @@ bool C2Compiler::is_intrinsic_supported(const methodHandle& method, bool is_virt
   case vmIntrinsics::_copyMemory:
     if (StubRoutines::unsafe_arraycopy() == NULL) return false;
     break;
+  case vmIntrinsics::_encodeAsciiArray:
+#if !defined(X86)
+    return false; // not yet implemented
+#endif
```

Comment:
It might be a little more work, but I think it's cleaner to move the decision whether the intrinisc is supported into the Matcher like for most other intrinsics and keep this code here platform independent. Otherwise we will get an increasing cascade of ifdefs as people start implementing this for other platforms.

Reply:
Not too much work. I recently introduced platform-specific `matcher_*.hpp` files then adding a boolean constant is easy (no need to much with the .ad files) and more or less free since such constants are no longer defined in a generated .cpp file which disables inlining and folding the constant.

Reply:
Does the changes in 9800a99 resolve your concerns?

Reply:
In principle yes, but shouldn't the condition read:
```cpp
if (!Matcher::match_rule_supported(Op_EncodeISOArray) || !Matcher::supports_encode_ascii_array) return false;
```
I.e. the intrinisc is supported if both conditions are true and not supported if either one of them is false?

Reply:
Resolved in 47633cb